lib/p2p/Pool.ts

```diff
@@ -647,12 +642,26 @@ class Pool extends EventEmitter {
       this.pendingOutgoingConnections.delete(peer.nodePubKey!);
     });
 
-    peer.once('close', () => {
+    peer.once('close', async () => {
+      // cleanup
       if (peer.nodePubKey) {
         this.pendingOutgoingConnections.delete(peer.nodePubKey);
         this.peers.remove(peer.nodePubKey);
       }
       this.emit('peer.close', peer);
+
+      // if handshake passed and peer disconnected from us for stalling or without specifying any reason -
+      // reconnect, for that might have been due to a temporary loss in connectivity
+      const legitReconnect =
+        peer.sentDisconnectionReason === undefined &&
+        (peer.recvDisconnectionReason === undefined || peer.recvDisconnectionReason === DisconnectionReason.ResponseStalling);
+      const addresses = peer.inbound ? peer.addresses : [peer.address];
```

Comment:
This looks like we won't attempt to reconnect using the peer's listening addresses if it's an outbound connection. Is that intentional?

If not, why not always use `peer.addresses` but additionally use `peer.address` as the `lastAddress` if the peer is outbound? So like make this line `const lastAddress = peer.inbound ? undefined : peer.address;`, make it `(addresses.length || lastAddress)` in the if statement below, and `const node = { addresses, lastAddress, nodePubKey: peer.nodePubKey };`.